lib/adb-device.js

```diff
@@ -214,6 +261,42 @@ class ADBDevice {
     await this.sendFile(source, destination, localId, remoteId);
   }
 
+  // runs ADB push to push a file to the device
+  async list (remotePath) {
+    let syncBuf = new Buffer("sync:.");
+    // open a SYNC stream on the device for pushing the file
+    let packet = await this.sendAndOkay(ADB_COMMANDS.CMD_OPEN
+                                       , 12345
+                                       , 0
+                                       , syncBuf);
+    // this is our local id, the devices remote id
+    let localId = packet.arg2;
+    // this is our remote id, the devices local id
+    let remoteId = packet.arg1;
+    packet = await this.stat(remotePath, localId, remoteId);
+    let fileType = packet.data.readUInt32LE(4) >>> 13;
+    if (fileType === FILE_TYPES.FILE || fileType === 0) { // remotePath is a file or doesn't exist
```

Comment:
I'm not sure if we even need to do any check here. The `LIST` command will just return an empty list if the file does not exist or if it is not a folder. Perhaps just skip this check? We don't check whether a symlink is a folder, so instead in code would have to check for both `output === []` and `output === -1`